aptible/resource_app.go

```diff
@@ -127,33 +145,42 @@ func resourceAppImport(d *schema.ResourceData, meta interface{}) ([]*schema.Reso
 
 // syncs Terraform state with changes made via the API outside of Terraform
 func resourceAppRead(d *schema.ResourceData, meta interface{}) error {
-	client := meta.(*providerMetadata).LegacyClient
-	appID := int64(d.Get("app_id").(int))
+	client := meta.(*providerMetadata).Client
+	appID := int32(d.Get("app_id").(int))
+	ctx := context.Background()
 
 	log.Println("Getting App with ID: " + strconv.Itoa(int(appID)))
 
-	app, err := client.GetApp(appID)
+	app, resp, err := client.AppsAPI.GetApp(ctx, appID).Execute()
 	if err != nil {
 		log.Println(err)
 		return err
 	}
-	if app.Deleted {
+	if resp.StatusCode == http.StatusNotFound {
 		d.SetId("")
 		return nil
 	}
```

Comment:
A 404 will return an error so if you want to detect it the status check has to go above the error check. e.g. https://github.com/aptible/terraform-provider-aptible/blob/c6cd94687339d2d70026d8fcc7e05b8a84a09a7b/aptible/resource_database.go#L182-L189

Reply:
ooh thanks!